lib/active_merchant/billing/gateways/global_collect.rb

```diff
@@ -277,9 +277,13 @@ def uri(action, authorization)
         end
       end
 
-      def commit(action, post, authorization = nil)
+      def idempotency_key(options)
+        "x-gcs-idempotence-key:#{options[:idempotency_key]}" if options[:idempotency_key]
+      end
+
+      def commit(action, post, authorization = nil, options = {})
```

Comment:
You did a nice job managing the addition of another argument with a default value to this method.

An alternative way to do this would be to use ruby's Keyword Arguments.

In this approach, the method could be defined like
```
def commit(action, post, authorization: nil, options: {})
```
where `authorization:` and `options:` uses the keyword syntax. Keyword Arguments are not positionally dependent, so this would allow you to set a default for both arguments AND ALSO let you call the method and pass `options` without needing to pass a value for `authorization`. So in `def authorize` you could call it like this:
```
commit(:authorize, post, options: options)
```
while in other places like `def capture` you can call it like this
```
commit(:capture, post, authorization: authorization)
```

This saves you from having to pass this annoying `nil` around. The downside is that you cannot pass values for the keyword arguments without using the keyword (i.e. you cannot call it like `commit(:capture, post, authorization)`), so you would have to change it everywhere you call it.

Maybe this would be a confusing pattern for some people. I don't see it used much in Active Merchant (although it is already being used in this file, with `def add_order`). Another judgement call that comes down to what you think is more readable and understandable.

My apologies if you were already familiar with keyword arguments and had already considered this approach.

Reply:
@dsmcclain I agreed with this and everything else so I pushed up the changes. Thanks again!